src/Bedrock.Framework.Experimental/Client/Pooling/EndPointPool.cs

```diff
@@ -91,17 +92,17 @@ private void IncrementConnectionCountNoLock()
             _connectionCount++;
         }
 
-        public ValueTask<ConnectionContext> GetConnectionAsync(CancellationToken cancellationToken = default)
+        public ValueTask<Connection> GetConnectionAsync(CancellationToken cancellationToken = default)
         {
             if (cancellationToken.IsCancellationRequested)
             {
-                return new ValueTask<ConnectionContext>(Task.FromCanceled<ConnectionContext>(cancellationToken));
+                return new ValueTask<Connection>(Task.FromCanceled<Connection>(cancellationToken));
```

Comment:
I believe this is in .NET 5:

```suggestion
                return ValueTask.FromCanceled<Connection>(cancellationToken);
```